src/main/java/io/github/bobocodebreskul/context/registry/AnnotatedBeanDefinitionReader.java

```diff
@@ -88,76 +84,44 @@ public void register(Class<?>... componentClasses) {
    * @param beanClass the class of the bean
    */
   public void registerBean(Class<?> beanClass) {
-    String name = extractBeanName(beanClass).orElse(null);
-    doRegisterBean(beanClass, name);
+    doRegisterBean(beanClass);
   }
 
-  /**
-   * Retrieves the associated Bean Definition Registry.
-   *
-   * @return The Bean Definition Registry to which this reader is operates on.
-   */
-  public final BeanDefinitionRegistry getBeanDefinitionRegistry() {
-    return beanDefinitionRegistry;
-  }
-
-  private Optional<String> extractBeanName(Class<?> beanClass) {
-    Set<String> componentAnnotations = Arrays.stream(beanClass.getAnnotations())
-        .filter(ReflectionUtils::isComponentAnnotation)
-        .map(annotation -> ReflectionUtils.getClassAnnotationValue(beanClass,
-            annotation.annotationType(), COMPONENT_NAME_FIELD, String.class))
-        .filter(beanName -> nonNull(beanName) && !beanName.isBlank())
-        .collect(Collectors.toSet());
-    if (componentAnnotations.isEmpty()) {
-      return Optional.empty();
-    } else if (componentAnnotations.size() == 1) {
-      return componentAnnotations.stream().findFirst();
-    }
-    String beanNames = String.join(", ", componentAnnotations);
-    log.error(
-        "For bean {} was found several different names definitions: [{}]. Please choose one.",
-        beanClass.getName(), beanNames);
-    throw new IllegalStateException(
-        UNCERTAIN_BEAN_NAME_EXCEPTION_MSG.formatted(beanClass.getName(), beanNames));
-  }
-
-  private <T> void doRegisterBean(Class<T> beanClass, String beanName) {
-    log.debug("doRegisterBean method invoked: beanClass={}, name={}",
-        beanClass.getName(),
-        beanName);
+  private <T> void doRegisterBean(Class<T> beanClass) {
+    log.debug("doRegisterBean method invoked: beanClass={}", beanClass);
     log.info("Registering the bean definition of class {}", beanClass.getName());
+    String name = BeanDefinitionReaderUtils.getBeanName(beanClass);
+
     // todo: create beanDefinitionValidator that validate things such as:
     //  bean name validity (not allowed characters), check for circular dependency
 
     var annotatedBeanDefinition = new AnnotatedGenericBeanDefinition(beanClass);
-    beanName = beanName != null ?
-        beanName : generateBeanName(annotatedBeanDefinition, beanDefinitionRegistry);
-    annotatedBeanDefinition.setName(beanName);
+    annotatedBeanDefinition.setName(name);
 
-    if (beanDefinitionRegistry.isBeanNameInUse(beanName)) {
+    if (beanDefinitionRegistry.isBeanNameInUse(name)) {
```

Comment:
We have removed this redundant check from here, but it still presens, probably it We have removed this redundant check from here, but it still presens, probably that's because of merge of our changes with develop branch. Need to be removed, since this check is present at the level of bean definition registry.

Reply:
fixed